rts/motoko-rts/src/bigint.rs

```diff
@@ -246,6 +246,25 @@ unsafe extern "C" fn bigint_of_int64(j: i64) -> SkewedPtr {
     persist_bigint(i)
 }
 
+#[cfg(feature = "ic")]
+#[no_mangle]
+unsafe extern "C" fn bigint_of_float64(j: f64) -> SkewedPtr {
+    let mut i = tmp_bigint();
+    mp_set_double(&mut i, j);
+    persist_bigint(i)
+}
+
+#[cfg(feature = "ic")]
+#[no_mangle]
+unsafe extern "C" fn bigint_to_float64(p: SkewedPtr) -> f64 {
```

Comment:
is `unsafe` needed?

Reply:
Yes, since I am calling another `unsafe` function. Is transitivity a curse of a blessing? :-)

Reply:
> is unsafe needed?

Good question. `unsafe` in this context means "memory safety not guaranteed". Dereferencing a raw pointer is unsafe because raw pointers are not borrow checked so you can read e.g. freed memory, or memory being accessed in another thread etc. We deal with raw pointers a lot in the RTS, it was easier to mark most functions as "unsafe". But if there are cases where any potential unsafety is checked (for example, by bounds checking in array access functions) then we can remove "unsafe" from the function type and use an unsafe block in the function body.

In the context of language runtimes, perhaps "safe if not buggy" is safe enough and we can move the `unsafe` from function types to function bodies, I'm not sure. `unsafe` does not affect code gen, it's just a signal to the programmer.

Reply:
Well, it works this way, so I'll leave it as-is. I was just wondering, and missing the intuition. I am fine now.